arangod/Graph/Graph.cpp

```diff
@@ -189,6 +190,58 @@ void Graph::addVertexCollection(std::string const& name) {
   _vertexColls.emplace(name);
 }
 
+void Graph::rebuildOrphans(EdgeDefinition const& oldEdgeDefinition) {
+  // previous orphans may still be orphans...
+  std::set<std::string> orphans{orphanCollections()};
+
+  // previous vertex collections from the overwritten may be orphaned...
+  setUnion(orphans, oldEdgeDefinition.getFrom());
+  setUnion(orphans, oldEdgeDefinition.getTo());
+
+  // ...except they occur in any other edge definition, including the new one.
+  for (auto const& it : edgeDefinitions()) {
+    EdgeDefinition const& edgeDef = it.second;
+
+    setMinus(orphans, edgeDef.getFrom());
+    setMinus(orphans, edgeDef.getTo());
+  }
+  _orphanColls = orphans;
+}
+
+void Graph::rebuildOrphans(EdgeDefinition const& oldEdgeDefinition,
+                           EdgeDefinition const& newEdgeDefinition) {
+    // previous orphans may still be orphans...
+    std::set<std::string> orphans{orphanCollections()};
+
+    // previous vertex collections from the overwritten may be orphaned...
+    setUnion(orphans, oldEdgeDefinition.getFrom());
+    setUnion(orphans, oldEdgeDefinition.getTo());
+
+    // ...except they occur in any other edge definition, including the new one.
+    for (auto const& it : edgeDefinitions()) {
+      std::string const& edgeCollection = it.first;
+
+      EdgeDefinition const& edgeDef =
+          edgeCollection == newEdgeDefinition.getName() ? newEdgeDefinition
+                                                        : it.second;
+
+      setMinus(orphans, edgeDef.getFrom());
+      setMinus(orphans, edgeDef.getTo());
+    }
+    _orphanColls = orphans;
```

Comment:
Please Revwrite.
Should first call the rebuildWith the oldEdgeDefinition, and afterwards remove those vertex collections that are part of the new definition from the orphansSet.

Reply:
Actually remove this implementation as it is superflous.